Fix Kerberos tests on JDK11:

eolivelli opened the pull request:
Descriptions of the changes in this PR:
- move to Kerby (MiniKDC code copied from Apache ZooKeeper repository)
- adapt tests to run with new KDC
- force TCP on krb5.conf for tests (this is needed for Kerby, which opens only TCP by default config)


Master Issue: #1710 

> ---
> In order to uphold a high standard for quality for code contributions, Apache BookKeeper runs various precommit
> checks for pull requests. A pull request can only be merged when it passes precommit checks. However running all
> the precommit checks can take a long time, some trivial changes don't need to run all the precommit checks. You
> can check following list to skip the tests that don't need to run for your pull request. Leave them unchecked if
> you are not sure, committers will help you:
>
> - [ ] [skip bookkeeper-server bookie tests]: skip testing `org.apache.bookkeeper.bookie` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server client tests]: skip testing `org.apache.bookkeeper.client` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server replication tests]: skip testing `org.apache.bookkeeper.replication` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server tls tests]: skip testing `org.apache.bookkeeper.tls` in bookkeeper-server module.
> - [ ] [skip bookkeeper-server remaining tests]: skip testing all other tests in bookkeeper-server module.
> - [ ] [skip integration tests]: skip docker based integration tests. if you make java code changes, you shouldn't skip integration tests.
> - [ ] [skip build java8]: skip build on java8. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> - [ ] [skip build java9]: skip build on java9. *ONLY* skip this when *ONLY* changing files under documentation under `site`.
> ---

> ---
> Be sure to do all of the following to help us incorporate your contribution
> quickly and easily:
>
> If this PR is a BookKeeper Proposal (BP):
>
> - [ ] Make sure the PR title is formatted like:
>     `<BP-#>: Description of bookkeeper proposal`
>     `e.g. BP-1: 64 bits ledger is support`
> - [ ] Attach the master issue link in the description of this PR.
> - [ ] Attach the google doc link if the BP is written in Google Doc.
>
> Otherwise:
> 
> - [ ] Make sure the PR title is formatted like:
>     `<Issue #>: Description of pull request`
>     `e.g. Issue 123: Description ...`
> - [ ] Make sure tests pass via `mvn clean apache-rat:check install spotbugs:check`.
> - [ ] Replace `<Issue #>` in the title with the actual Issue number.
> 
> ---

sijie commented on bookkeeper-server/src/test/java/org/apache/bookkeeper/sasl/MiniKdc.java:
```diff
@@ -0,0 +1,356 @@
+ * The generated krb5.conf forces TCP connections.
+ */
+/*
+ * This code is originally from HDFS, see the file name MiniKdc there
```
if it is LICENSE v2, it should be okay.

ivankelly replied:
If it comes from an ASF project, it's already covered by our notice. However, if hadoop brought it in from somewhere else, they should have that noted in their notice, and we should bubble that up.

ivankelly replied:
Looks to be 100% original hadoop code, so there's no change needed.

sijie commented:
@eolivelli please update the license and notice files.

eolivelli commented:
@sijie 
I think that this line covers this change:

> This product includes software developed at
> The Apache Software Foundation (http://www.apache.org/).

You can find it in NOTICE-all-bin.txt and NOTICE-server-bin.txt
https://github.com/apache/bookkeeper/blob/443b24a630500cd248041de4583f09dbce90c398/bookkeeper-dist/src/main/resources/NOTICE-all.bin.txt#L4

eolivelli commented:
retest this please

eolivelli commented:
@sijie can you take a look ? So that I can move forward. Thanks

ivankelly commented on bookkeeper-server/src/test/java/org/apache/bookkeeper/sasl/MiniKdc.java:
```diff
@@ -0,0 +1,356 @@
+ * The generated krb5.conf forces TCP connections.
+ */
+/*
+ * This code is originally from HDFS, see the file name MiniKdc there
```
Add a link to the repo for convenience.

eolivelli replied:
done

eolivelli commented:
Thank you @ivankelly I have fixed checkstyle and added the link to current MiniKdc from HDFS.

I will commit as soon as I have all green lights from CI

ivankelly commented:
:+1:

eolivelli commented:
retest this please

eolivelli commented:
rerun bookkeeper-server replication tests

eolivelli commented:
Travis is failing for LICENSE issues:
$BK_HOME/dev/check-binary-license ./bookkeeper-dist/all/target/bookkeeper-all-*-bin.tar.gz;
com.github.stephenc.jcip-jcip-annotations-1.0-1.jar unaccounted for in LICENSE
com.nimbusds-nimbus-jose-jwt-4.41.2.jar unaccounted for in LICENSE
net.minidev-accessors-smart-1.2.jar unaccounted for in LICENSE
net.minidev-json-smart-2.3.jar unaccounted for in LICENSE
org.apache.kerby-kerb-admin-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-client-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-common-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-core-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-crypto-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-identity-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-server-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-simplekdc-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerb-util-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerby-asn1-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerby-config-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerby-pkix-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerby-util-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-kerby-xdr-1.1.1.jar unaccounted for in LICENSE
org.apache.kerby-token-provider-1.1.1.jar unaccounted for in LICENSE
org.ow2.asm-asm-5.0.4.jar unaccounted for in LICENSE

It looks like there are issues with the LICENSE/NOTICE.
See http://bookkeeper.apache.org/community/licensing for details on how to fix.

sijie commented:
> Travis is failing for LICENSE issues:
$BK_HOME/dev/check-binary-license ./bookkeeper-dist/all/target/bookkeeper-all-*-bin.tar.gz;

yes. that was what I meant "fixing license/notice issue"

sijie commented:
run integration tests

eolivelli commented:
@sijie got it. thanks

sijie commented:
run integration tests

sijie commented:
run integration tests
